subprojects/analysis-kotlin-descriptors/ide/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/ide/IdeCompilerExtensionPointProvider.kt

```diff
@@ -1,46 +1,8 @@
 package org.jetbrains.dokka.analysis.kotlin.descriptors.ide
 
 import org.jetbrains.dokka.analysis.kotlin.descriptors.compiler.CompilerExtensionPointProvider
-import org.jetbrains.kotlin.caches.resolve.CommonPlatformKindResolution
-import org.jetbrains.kotlin.caches.resolve.IdePlatformKindResolution
-import org.jetbrains.kotlin.caches.resolve.JsPlatformKindResolution
-import org.jetbrains.kotlin.caches.resolve.JvmPlatformKindResolution
-import org.jetbrains.kotlin.extensions.ApplicationExtensionDescriptor
-import org.jetbrains.kotlin.ide.konan.NativePlatformKindResolution
-import org.jetbrains.kotlin.platform.IdePlatformKind
-import org.jetbrains.kotlin.platform.impl.CommonIdePlatformKind
-import org.jetbrains.kotlin.platform.impl.JsIdePlatformKind
-import org.jetbrains.kotlin.platform.impl.JvmIdePlatformKind
-import org.jetbrains.kotlin.platform.impl.NativeIdePlatformKind
 
-internal class IdeCompilerExtensionPointProvider : CompilerExtensionPointProvider {
-    override fun get(): List<CompilerExtensionPointProvider.CompilerExtensionPoint> {
-
-        @Suppress("UNCHECKED_CAST")
-        val idePlatformKind = CompilerExtensionPointProvider.CompilerExtensionPoint(
-            ApplicationExtensionDescriptor(
-                "org.jetbrains.kotlin.idePlatformKind",
-                IdePlatformKind::class.java
-            ) as ApplicationExtensionDescriptor<Any>,
-            listOf(
-                CommonIdePlatformKind,
-                JvmIdePlatformKind,
-                JsIdePlatformKind,
-                NativeIdePlatformKind
-            )
-        )
 
-        @Suppress("UNCHECKED_CAST")
-        val resolution = CompilerExtensionPointProvider.CompilerExtensionPoint(
```

Comment:
In K1 `IdePlatformKindResolution` is used to configure global caches to analyze code for different platforms. It contains platform factories, they are used in `IdeaResolverForProject` and in the inheritors of `ResolverForModuleFactory` for different platforms to create some components for the frontend: biltins, package providers for Klib'as and fabrics for creating classes...
But `IdeaResolverForProject` is not used in Dokka.
`KotlinCacheService` we have our own [stub](https://github.com/Kotlin/dokka/blob/12a386bb7185f862a1cbd831e6856c4235953833/kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/CoreKotlinCacheService.kt#L13)
`ResolverForModuleFactory` we have our own [ones](https://github.com/Kotlin/dokka/blob/12a386bb7185f862a1cbd831e6856c4235953833/kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/resolve/DokkaNativeResolverForModuleFactory.kt#L61), where `PlatformKindResolution` is not called.